### PR TITLE
Add get vertex composite index support to grpc

### PR DESCRIPTION
**Pull request by farodin91**
* Get vertex composite index by name
* Get all vertex composite indices

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java [hunk 1]
```diff
@@ -507,4 +509,183 @@ public void testGetEdgeLabelByNameEdgeLabelWithMultipleEdgeProperties(int number
...
+            .setIndexOnly(vertex)
+            .addKeys(property));
+
+        VertexCompositeGraphIndex index = schemaManagerClient.getVertexCompositeGraphIndexByName(name);
+
+        assertEquals(name, index.getName());
+        assertEquals(1, index.getKeysCount());
+        assertEquals("testProperty", index.getKeys(0).getName());
+        assertFalse(index.getUnique());
+        assertTrue(index.hasIndexOnly());
```
Comment (FlorianHockmann) [inline janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java:623]: Wouldn't it also be possible to check the vertex label used here?

File: janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java [hunk 2]
```diff
@@ -507,4 +509,183 @@ public void testGetEdgeLabelByNameEdgeLabelWithMultipleEdgeProperties(int number
...
+
+        for (int i = 0; i < numberOfVertices; i++) {
+            VertexCompositeGraphIndex.Builder indexBuilder = VertexCompositeGraphIndex.newBuilder()
+                .setName("testGetVertexCompositeGraphIndices" + i)
+                .addKeys(test);
+            Object id = createVertexCompositeGraphIndex(defaultGraphName, indexBuilder);
+                createdIds.add(id);
+        }
+
+        List<VertexCompositeGraphIndex> vertexLabels = schemaManagerClient.getVertexCompositeGraphIndices();
```
Comment (FlorianHockmann) [inline janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java:683]:
```suggestion
        List<VertexCompositeGraphIndex> compositeGraphIndices = schemaManagerClient.getVertexCompositeGraphIndices();
```

File: janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerProvider.java [hunk 3]
```diff
@@ -60,4 +63,28 @@ public List<EdgeLabel> getEdgeLabels() {
...
+        JanusGraphIndex graphIndex = management.getGraphIndex(indexName);
+        if (graphIndex == null) {
+            return null;
+        }
+        if (!Vertex.class.isAssignableFrom(graphIndex.getIndexedElement())) {
+            return null;
+        }
+
+        if (!graphIndex.isCompositeIndex()) {
+            return null; //error?
```
Comment (FlorianHockmann) [inline janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerProvider.java:77]: I also think that an error would make more sense than just returning `null` which could also mean that no index was found. If no index was found, a user might try next to create it which will fail if it already exists.
This should also be tested.

File: janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerProvider.java [hunk 4]
```diff
@@ -60,4 +63,28 @@ public List<EdgeLabel> getEdgeLabels() {
...
             .map(GrpcUtils::createEdgeLabelProto).collect(Collectors.toList());
     }
+
+    public VertexCompositeGraphIndex getVertexCompositeGraphIndexByName(String indexName) {
+        JanusGraphIndex graphIndex = management.getGraphIndex(indexName);
+        if (graphIndex == null) {
+            return null;
+        }
+        if (!Vertex.class.isAssignableFrom(graphIndex.getIndexedElement())) {
+            return null;
```
Comment (FlorianHockmann) [inline janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerProvider.java:73]: Same here, I think an error would be better as this method shouldn't be called for an edge index. (Should also be tested.)

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 5]
```diff
@@ -85,6 +85,19 @@ message VertexLabel {
...
 }
 
+message VertexCompositeGraphIndex {
+    // The Internal id.
+    google.protobuf.Any id = 1;
+    // The GraphIndex Name.
+    string name = 2;
+    // keys.
+    repeated VertexProperty keys = 3;
+    // Only for following index.
```
Comment (FlorianHockmann) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:95]:
```suggestion
    // Only for following vertex label.
```

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 6]
```diff
@@ -85,6 +85,19 @@ message VertexLabel {
     repeated VertexProperty properties = 5;
 }
 
+message VertexCompositeGraphIndex {
+    // The Internal id.
+    google.protobuf.Any id = 1;
+    // The GraphIndex Name.
+    string name = 2;
+    // keys.
```
Comment (FlorianHockmann) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:93]:
```suggestion
    // The vertex property keys.
```

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 7]
```diff
@@ -85,6 +85,19 @@ message VertexLabel {
...
+    repeated VertexProperty keys = 3;
+    // Only for following index.
+    VertexLabel index_only = 4; //nullable
+    // Marking as unique.
```
Comment (FlorianHockmann) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:97]:
```suggestion
    // Whether the index is marked as unique or not.
```

File: janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerImpl.java [hunk 8]
```diff
@@ -122,4 +123,41 @@ public void getEdgeLabels(
...
+        final String indexName = request.getName();
+        if (indexName.isEmpty()) {
+            responseObserver.onError(Status.INVALID_ARGUMENT
+                .withDescription("name is required").asException());
+            return;
+        }
+        VertexCompositeGraphIndex index = provider.getVertexCompositeGraphIndexByName(indexName);
+        if (index == null) {
+            responseObserver.onError(Status.NOT_FOUND
+                .withDescription("No vertexLabel found with name: " + indexName).asException());
```
Comment (FlorianHockmann) [inline janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerImpl.java:144]:
```suggestion
                .withDescription("No composite graph index found with name: " + indexName).asException());
```

**Review by FlorianHockmann — COMMENTED** [review]
Apart from my inline comments, is there an issue for all these GRPC PRs? Would be good to see which parts are already done and which are still missing.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/types/CompositeIndexType.java [hunk 9]
```diff
@@ -23,6 +23,11 @@
 */
 public interface CompositeIndexType extends IndexType {
 
+    /**
+     * @deprecated use longId()
+     * @return return index id
+     */
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (farodin91) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/types/CompositeIndexType.java:30]: My plan is to make the naming more consistent.

**Comment by farodin91** [comment]
> Apart from my inline comments, is there an issue for all these GRPC PRs? Would be good to see which parts are already done and which are still missing.

https://github.com/JanusGraph/janusgraph/issues/1853

**Comment by farodin91** [comment]
@FlorianHockmann Would you like to review it again?

**Review by FlorianHockmann — APPROVED** [review]

**Pull request merged**